Mlem/App/Utility/Extensions/Content Models/Interactable1Providing+Extensions.swift

```diff
@@ -56,6 +56,12 @@ extension Interactable1Providing {
                 HapticManager.main.play(haptic: .success, priority: .low)
             }
             self2.toggleSaved()
+            
+            @Setting(\.upvoteOnSave) var upvoteOnSave
+            if upvoteOnSave, self2.votes.myVote != .upvote {
+                self2.updateVote(.upvote)
+            }
+            
```

Comment:
Need to check whether we're saving or un-saving otherwise the post is upvoted on unsave

```suggestion            
            @Setting(\.upvoteOnSave) var upvoteOnSave
            if upvoteOnSave, !self2.saved {
                self2.updateVote(.upvote)
            }
            
            self2.toggleSaved()     
```